lib/routes.js

```diff
@@ -35,8 +36,9 @@ const statsClient = new StatsClient(localCacheClient, STATS_INTERVAL,
     STATS_EXPIRY);
 
 function checkUnsuportedRoutes(req, res, log) {
-    if (req.query.policy !== undefined ||
-        req.query.tagging !== undefined) {
+    const unsupportedRequestQuery = Object.keys(req.query)
```

Comment:
Since this would happen on every request, it would be better if we test the url vs looping through the properties of query object (as it is expensive). It's a common attack vector for public facing apps.
Here's what I would suggest
```js
const reqQueryStr = url.parse(req.url).query;
unsupportedQueries.some(q => reqQueryStr.includes(q));
```
Since we already parse the url once in https://github.com/scality/S3/blob/master/lib/utils.js#L252, to avoid parsing multiple times you could
```js
const parsedUrl = url.parse(request.url);
request.queryStr = parsedUrl.query;
request.query = querystring.parse(parsedUrl.query);
```